### PR TITLE
Add PUBLIC_IP_SERVICE to netmaker (server).

### PR DESCRIPTION
**Pull request by cameronts**
To test: define environment variable PUBLIC_IP_SERVICE to the service to check first before running netmaker.

### CODE REVIEW

File: servercfg/serverconf.go [hunk 1]
```diff
@@ -427,7 +428,16 @@ func GetPublicIP() (string, error) {
 	var err error
 
 	iplist := []string{"https://ip.server.gravitl.com", "https://ifconfig.me", "https://api.ipify.org", "https://ipinfo.io/ip"}
+	publicIpService := os.Getenv("PUBLIC_IP_SERVICE")
+	if publicIpService != "" {
+		fmt.Println("User provided public IP service is", publicIpService)
```
Comment (mattkasun) [inline servercfg/serverconf.go:433]: Use logger.Log()

File: servercfg/serverconf.go [hunk 2]
```diff
@@ -427,7 +428,16 @@ func GetPublicIP() (string, error) {
...
+
+		// prepend the user-specified service so it's checked first
+		iplist = append([]string{publicIpService}, iplist...)
+	}
+
 	for _, ipserver := range iplist {
+		fmt.Println("Running public IP check with service", ipserver)
```
Comment (mattkasun) [inline servercfg/serverconf.go:440]: logger.Log()

File: servercfg/serverconf.go [hunk 3]
```diff
@@ -442,6 +452,7 @@ func GetPublicIP() (string, error) {
 				continue
 			}
 			endpoint = string(bodyBytes)
+			fmt.Println("Public IP address is", endpoint)
```
Comment (mattkasun) [inline servercfg/serverconf.go:455]: lighter.Log()

File: servercfg/serverconf.go (hunk 1 above)
Comment (mattkasun) [inline servercfg/serverconf.go:433]: Do we need to validate that publicIpService is a valid url?

**Review by mattkasun — CHANGES_REQUESTED** [review]
Change fmt.Println to logger.Log

File: servercfg/serverconf.go (hunk 1 above)
Comment (cameronts) [reply servercfg/serverconf.go:433]: Thanks Matt, would like to.  We're working on how to get that done.

File: servercfg/serverconf.go (hunk 2 above)
Comment (cameronts) [reply servercfg/serverconf.go:440]: Thanks Matt, would like to. We're working on how to get that done.

File: servercfg/serverconf.go (hunk 3 above)
Comment (cameronts) [reply servercfg/serverconf.go:455]: Thanks Matt, would like to. We're working on how to get that done.

File: servercfg/serverconf.go (hunk 1 above)
Comment (0xdcarns) [reply servercfg/serverconf.go:433]: I assume if it fails, it would move on to the predefined ones

**Comment by cameronts** [comment]
Yes it did in my testing

File: servercfg/serverconf.go (hunk 1 above)
Comment (cameronts) [reply servercfg/serverconf.go:433]: > I assume if it fails, it would move on to the predefined ones

Yes it did in my testing.

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**